homeassistant/auth/providers/__init__.py

```diff
@@ -223,7 +223,7 @@ async def async_step_select_mfa_module(
             errors["base"] = "invalid_auth_module"
 
         if len(self.available_mfa_modules) == 1:
-            self._auth_module_id = list(self.available_mfa_modules)[0]
+            self._auth_module_id = next(iter(self.available_mfa_modules))
```

Comment:
I'm not sure we should really do that. Of course `next` might be slightly faster but in most of these cases that doesn't really matter all that much. The more important thing IMO is that, especially for beginners, wrapping something with `list` and accessing the first element is easier to reason about than `next(...)`.

_If it makes sense in a specific case, e.g. because it's part of the core and often executed, I'm not against using `next` but I would suggest we don't enable this rule._

Reply:
Okay, then I revert the changes and put it on ignore instead.